go/stellar/bundle/boxer.go

```diff
@@ -171,17 +176,34 @@ func Unbox(decodeRes DecodeResult, visibleBundleB64 string,
 // Does not check invariants.
 func Decrypt(encBundle stellar1.EncryptedBundle,
 	puk libkb.PerUserKeySeed) (res stellar1.BundleSecretVersioned, err error) {
+	return decrypt2(encBundle, false, puk)
+}
+
+// CORE-8135 phase 2: collapse into Decrypt
+func decrypt2(encBundle stellar1.EncryptedBundle, supportCryptV1 bool,
+	puk libkb.PerUserKeySeed) (res stellar1.BundleSecretVersioned, err error) {
+	switch encBundle.V {
+	case 1:
+		// CORE-8135 phase 2: remove v1 decryption support
+		if !supportCryptV1 {
+			return res, fmt.Errorf("stellar secret bundle encryption version 1 has been retired. Try `keybase wallet fixup`")
+		}
+	case 2:
+	default:
+		return res, fmt.Errorf("unsupported stellar secret bundle encryption version: %v", encBundle.V)
+	}
+
 	// Derive key
-	// TODO: We're passing the wrong context in here. We need to migrate to libkb.DeriveReasonPUKStellarBundle. (CORE-8135)
-	symmetricKey, err := puk.DeriveSymmetricKey(libkb.DeriveReasonPUKPrev)
+	reason := libkb.DeriveReasonPUKStellarBundle
+	if encBundle.V == 1 {
+		reason = libkb.DeriveReasonPUKPrev
```

Comment:
Maybe another comment here just to say what's up?

Reply:
This code will be gone quicker than you can say https://github.com/keybase/client/pull/12496